private/bufpkg/bufplugin/bufpluginconfig/bufpluginconfig.go

```diff
@@ -0,0 +1,205 @@
+	// V1Version is the version string used to indicate the v1 version of the buf.plugin.yaml file.
+	V1Version = "v1"
+)
+
+var (
+	// AllConfigFilePaths are all acceptable config file paths without overrides.
+	//
+	// These are in the order we should check.
+	AllConfigFilePaths = []string{
+		ExternalConfigFilePath,
+	}
+)
+
+// Config is the plugin config.
+type Config struct {
+	// Name is the name of the plugin (e.g. 'buf.build/protocolbuffers/go').
+	Name bufpluginref.PluginIdentity
+	// Options is the set of options available to the plugin.
+	//
+	// For now, all options are string values. This could eventually
+	// support other types (like JSON Schema and Terraform variables),
+	// where strings are the default value unless otherwise specified.
+	//
+	// Note that some legacy plugins don't always express their options
+	// as key value pairs. For example, protoc-gen-java has an option
+	// that can be passed like so:
+	//
+	//  java_opt=annotate_code
+	//
+	// In those cases, the option value in this map will be set to
+	// the empty string, and the option will be propagated to the
+	// compiler without the '=' delimiter.
+	Options map[string]string
+	// Runtime is the runtime configuration, which lets the user specify
+	// runtime dependencies, and other metadata that applies to a specific
+	// remote generation registry (e.g. the Go module proxy, NPM registry,
+	// etc).
+	Runtime *RuntimeConfig
+}
+
+// RuntimeConfig is the configuration for the runtime of a plugin.
+//
+// Only one field will be set.
+type RuntimeConfig struct {
+	Go      *GoRuntimeConfig
+	NPM     *NPMRuntimeConfig
+	Archive *ArchiveRuntimeConfig
+}
+
+// GoRuntimeConfig is the runtime configuration for a Go plugin.
+type GoRuntimeConfig struct {
+	MinVersion string
+	Deps       []string
+}
+
+// NPMRuntimeConfig is the runtime configuration for a JavaScript NPM plugin.
+type NPMRuntimeConfig struct {
+	Deps []string
+}
+
+// ArchiveRuntimeConfig is the runtime configuration for a plugin that can be downloaded
+// as an archive instead of a language-specific registry.
+type ArchiveRuntimeConfig struct {
+	Deps []string
+}
+
+// GetConfigForBucket gets the Config for the YAML data at ConfigFilePath.
+//
+// If the data is of length 0, returns the default config.
+func GetConfigForBucket(ctx context.Context, readBucket storage.ReadBucket) (*Config, error) {
+	return getConfigForBucket(ctx, readBucket)
+}
+
+// GetConfigForData gets the Config for the given JSON or YAML data.
+//
+// If the data is of length 0, returns the default config.
+func GetConfigForData(ctx context.Context, data []byte) (*Config, error) {
+	return getConfigForData(ctx, data)
+}
+
+// ExistingConfigFilePath checks if a configuration file exists, and if so, returns the path
+// within the ReadBucket of this configuration file.
+//
+// Returns empty string and no error if no configuration file exists.
+func ExistingConfigFilePath(ctx context.Context, readBucket storage.ReadBucket) (string, error) {
```

Comment:
Do we need this?

Reply:
For now, no - I'll remove it and we can add it back later if we ever do.